### ISSUE TITLE
volume rendering camera's north_vector should be a property

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1642]
Right now several of the camera attributes are implemented as properties (e.g `camera.position`, `camera.focus`, and `camera.resolution`). We should also make `camera.north_vector` a property that has a callback that fires when the `north_vector` is set to ensure that the camera's orientation gets updated when the `north_vector` gets set. Right now you need to do something like:

```
sc.camera.north_vector = [0, 0, 1]
sc.camera.switch_orientation()
```

It would be nice if you just needed to do:

```
sc.camera.north_vector = [0, 0, 1]
```

like the other camera attributes that are implemented as properties.

**Comment by pranjaltale16** [comment]
Hello, I would like to work on this issue.
I think, we can add something like this to make this work.

```python
def north_vector():
  doc = '''

  '''
  
  def fget(self):
    return selt.north_vector

  def fset(self, value):
    _north_vector =  _sanitize_camera_property_units(value, self.scene)
    self.north_vector = _north_vector
    self.switch_orientation()

  def fdel(self):
    del self.north_vector
    self.north_vector = None
        return locals()
    north_vector = property(**north_vector())
```
I'm not sure about fset and whether we  should shift north_vector out of ```__init__()``` or not.
Could you please help me with this?

**Comment by ngoldbaum** [comment]
You need to define a `_north_vector` attribute (just like the other properties). See:

https://github.com/yt-project/yt/blob/f554a60d5b73704fa99a2579ccf66ad0370dd811/yt/visualization/volume_rendering/camera.py#L116-L120

You'll also need to make sure that it gets initialized properly.

Do you have a test script? You'll want to write a script that works with the current version of yt and then use that script to debug with as you modify yt's internals.

**Comment by pranjaltale16** [comment]
I don't have a script yet. I don't have much idea how should I write that script. Can you help me with that?

**Comment by ngoldbaum** [comment]
Start with the documentation for the volume renderer:

http://yt-project.org/docs/dev/visualizing/volume_rendering.html

or perhaps one of the volume rendering scripts in the cookbook:

http://yt-project.org/docs/dev/cookbook/index.html

All of the code in the documentation should be runnable, although you might need to download a test dataset from yt-project.org/data. 

Try modifying a script from the documentation to set a new `north_vector`. Make sure that works with the current version of yt. Then apply your code modifications and test again. Make sure that the result doesn't change and make sure that your new property makes it so you don't need to manually call `camera.switch_orientation()` after changing the `north_vector`.

**Comment by pranjaltale16** [comment]
Yeah, I went through the documentation and cookbook. This will take time for me(since I'm completely new) I will hack around something else and fix this issue on weekend. Is it okay?

**Comment by ngoldbaum** [comment]
You don't need to check in with us, please work on whatever you feel like at whatever pace you want to work.
